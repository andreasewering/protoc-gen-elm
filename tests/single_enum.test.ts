const {
  runPlugin,
  compileElm,
  getGeneratedFileContents,
} = require("./snapshot_test_base");

describe("protoc-gen-elm", () => {
  beforeAll(() => runPlugin("single_enum.proto"));
  const expectedElmFileName = "Proto/SingleEnum.elm";

  it("generates a valid elm file for single_enum.proto", async () => {
    await compileElm(expectedElmFileName);
  });

  it("generates the expected code for single_enum.proto", async () => {
    const generatedContent = await getGeneratedFileContents(expectedElmFileName);
    expect(generatedContent).toMatchSnapshot();
  });
});
