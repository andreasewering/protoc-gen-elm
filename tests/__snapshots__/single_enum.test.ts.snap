// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`protoc-gen-elm generates the expected code for single_enum.proto 1`] = `
"{- !!! DO NOT EDIT THIS FILE MANUALLY !!! -}


module Proto.SingleEnum exposing
    ( AnEnum(..)
    , anEnumDecoder
    , toAnEnumEncoder
    )

{-| ProtoBuf module: \`Proto.SingleEnum\`

This module was generated automatically using

  - [\`protoc-gen-elm\`](https://www.npmjs.com/package/protoc-gen-elm) 1.0.0
  - \`protoc\` 3.14.0
  - the following specification file: \`single_enum.proto\`

To run it use [\`elm-protocol-buffers\`](https://package.elm-lang.org/packages/eriktim/elm-protocol-buffers/1.1.0) version 1.1.0 or higher.


# Model

@docs AnEnum


# Decoder

@docs anEnumDecoder


# Encoder

@docs toAnEnumEncoder

-}

import Protobuf.Decode as Decode
import Protobuf.Encode as Encode



-- MODEL


{-| \`AnEnum\` enumeration
-}
type AnEnum
    = OptionA
    | OptionB
    | OPTIONC
    | AnEnumUnrecognized_ Int



-- DECODER


anEnumDecoder : Decode.Decoder AnEnum
anEnumDecoder =
    Decode.int32
        |> Decode.map
            (\\\\value ->
                case value of
                    0 ->
                        OptionA

                    1 ->
                        OptionB

                    2 ->
                        OPTIONC

                    v ->
                        AnEnumUnrecognized_ v
            )



-- ENCODER


toAnEnumEncoder : AnEnum -> Encode.Encoder
toAnEnumEncoder value =
    Encode.int32 <|
        case value of
            OptionA ->
                0

            OptionB ->
                1

            OPTIONC ->
                2

            AnEnumUnrecognized_ v ->
                v



-- SETTERS



"
`;
